h/services/delete_group.py

```diff
@@ -29,11 +28,10 @@ def _delete_annotations(self, group):
             raise DeletePublicGroupError("Public group can not be deleted")
 
         annotations = self.request.db.query(Annotation).filter_by(groupid=group.pubid)
-        for annotation in annotations:
-            storage.delete_annotation(self.request.db, annotation.id)
-            event = AnnotationEvent(self.request, annotation.id, "delete")
-            self.request.notify_after_commit(event)
+        for ann in annotations:
+            self._annotation_delete_service.delete(ann)
```

Comment:
I guess `delete_annotations(iterable_of_annotations)` could be an `AnnotationDeleteService` method. Then in future we wouldn't need to duplicate this loop each time something wants to delete multiple annotations.

Reply:
This is a good idea. I think there are probably some candidates for this in other services as well.